### ISSUE TITLE
TypeError: must be str, not NoneType

### ISSUE THREAD

**Issue by cookiengineer** [kryptxy/torrench#11]
When I use torrench on my 3G connection, I cannot influence which proxies it selects. This is bad and there should be an option to do so. Some proxies seem to ship different HTML than others (I guess advertising stuff). Also, when an HTTP connection is used, 3G carriers always inject their own HTML, replace image urls etc. pp. - so HTTPS should be enforced.

The error message I'm getting is:

```bash
Obtaining proxies...
>>> https://pirateproxy.cam
>>> https://thepiratebay.red
>>> https://thehiddenbay.info

>>> Using https://pirateproxy.cam
Fetching results...
Traceback (most recent call last):
  File "/usr/bin/torrench", line 11, in <module>
    load_entry_point('torrench==1.0.3', 'console_scripts', 'torrench')()
  File "/usr/lib/python3.6/site-packages/torrench/__main__.py", line 31, in main
    resolve_args(args)
  File "/usr/lib/python3.6/site-packages/torrench/__main__.py", line 67, in resolve_args
    Torrench.main(input_title, page_limit, 1) ## 1 means TPB
  File "/usr/lib/python3.6/site-packages/torrench/Torrench.py", line 21, in main
    tpb.main(title, page)
  File "/usr/lib/python3.6/site-packages/torrench/tpb/main.py", line 130, in main
    name = GREEN + name + RESET
TypeError: must be str, not NoneType
```

The TL;DR is: Please make it possible to manually select a piratebay proxy via flag, so that this can be prevented.

**Comment by kryptxy** [comment]
Please update to latest version (v1.0.42). This issue is resolved in latest version. If you still get this error, please provide OS in use, along with exact search string that gives this error.

**Comment by kryptxy** [comment]
Closing assuming its resolved. Let me know if it isn't along with details as mentioned above.

**Comment by cookiengineer** [comment]
I updated to HEAD (using AUR) and it seems to be resolved.
Sorry for the bug report, I should've checked for updates first ;(

I'll reopen this issue in case my carrier's 3G connection was the cause (it's kinda hard to reproduce with the failed/closed connections). So far testing it over the weekend produced no error.

**Comment by cookiengineer** [comment]
Damn, now I get still an error. BUT I guess it's more related to the pirateproxy.cam than to other proxies:

```bash
######################
#                    #
#    ThePirateBay    #
#                    #
######################

Obtaining proxies...
>>> Trying https://pirateproxy.cam
Fetching results...
Traceback (most recent call last):
  File "/usr/bin/torrench", line 11, in <module>
    load_entry_point('torrench==1.0.42', 'console_scripts', 'torrench')()
  File "/usr/lib/python3.6/site-packages/torrench/__main__.py", line 8, in main
    args()
  File "/usr/lib/python3.6/site-packages/torrench/Torrench.py", line 26, in args
    main(args)
  File "/usr/lib/python3.6/site-packages/torrench/Torrench.py", line 87, in main
    torrent_fetch(args)
  File "/usr/lib/python3.6/site-packages/torrench/Torrench.py", line 55, in torrent_fetch
    tpb.main(input_title, page_limit)
  File "/usr/lib/python3.6/site-packages/torrench/modules/thepiratebay.py", line 294, in main
    output = get_results(url, title, page_limit)
  File "/usr/lib/python3.6/site-packages/torrench/modules/thepiratebay.py", line 114, in get_results
    output = fetch_results(soup)
  File "/usr/lib/python3.6/site-packages/torrench/modules/thepiratebay.py", line 141, in fetch_results
    website_url = soup.find('img', id='TPBlogo')['src'].split('/')[2]
TypeError: 'NoneType' object is not subscriptable
```

**Comment by cookiengineer** [comment]
When I try out the pirateproxy.cam, it gives me a 504 - Gateway Timeout very often. Maybe this is not caught as an error in the code?

**Comment by kryptxy** [comment]
Similar issue is already open.
https://github.com/kryptxy/torrench/issues/16

Yes. Next update will be able to handle such errors more efficiently.
